File: frontend/src/utils/DBConnection.js
let mysql = require('mysql')

export default class DBConnection{
    constructor(variant){
        //to be implemented...
        this.connection = undefined
    }
    connect = (host,user,pwd,db)=>{
        this.connection = mysql.createConnection({
            host:host,
            user:user,
            password:pwd,
            database:db
        })
        console.log(this.connection.state)
        if(this.connection){
            this.connection.connect()
            console.log(this.connection.state)
        }
    }
    

}

File: backend/dataapp/DBConnection.py
from audioop import add
from encodings import utf_8
from mimetypes import init
from select import select
# import sqlalchemy
import pymysql,pymysql.cursors

class DBConnection:
    def __init__(self, user, pwd, address, dbname, port=3306, dbtype='mysql', driver='pymysql') -> None:
        self.conn = pymysql.connect(host=address, port=port, user = user, password=pwd, database=dbname, charset='utf8', ssl={'ssl':{}})
        # self.cursor = self.conn.cursor()
        

    def test_fetch(self):
        sql = 'select * from stu;'
        cursor = self.conn.cursor(cursor=pymysql.cursors.DictCursor)
        cursor.execute(sql)
        ret = cursor.fetchall()
        print(ret)
        return ret
